core/src/main/java/org/axonframework/serialization/MessageSerializer.java

```diff
@@ -60,9 +60,6 @@ public MessageSerializer(Serializer serializer) {
      */
     public static <T> SerializedObject<T> serializePayload(Message<?> message, Serializer serializer,
```

Comment:
As the `Message` is now `SerializationAware`, shouldn't we simply return `message.serializePayload(serializer, expectedRepresentation)` directly?

Reply:
Hah, no. We'll get `StackoverflowException` since serialization for the `Message` will eventually call the `MessageSerializer` which would call `message.serialize...`, you get the point ;)

Reply:
Aaah I see that the `SerializedObjectHolder`, which you perform the `serializePayload()` and `serializeMetaData()` function against, internally uses the `MessageSerializer.serializePayload()`/`MessageSerializer.serializeMetaData()`...ok, get it, thanks for clarifying it!